de-lib/src/main/java/org/iplantc/de/client/services/impl/GroupServiceFacadeImpl.java

```diff
@@ -410,15 +421,14 @@ public void denyRequestToJoinTeam(Group team,
     }
 
     @Override
-    public void searchTeams(String searchTerm, AsyncCallback<List<Group>> callback) {
-        String address = TEAMS + "?search=" + URL.encodeQueryString(searchTerm);
+    public void searchTeams(String searchTerm, AsyncCallback<Splittable> callback) {
+        String address = TEAMS + "?search=" + URL.encodeQueryString(searchTerm) + "&details=true";
 
         ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, address);
-        deService.getServiceData(wrapper, new AsyncCallbackConverter<String, List<Group>>(callback) {
+        deService.getServiceData(wrapper, new AsyncCallbackConverter<String, Splittable>(callback) {
             @Override
-            protected List<Group> convertFrom(String object) {
-                GroupList groupList = AutoBeanCodex.decode(factory, GroupList.class, object).as();
-                return groupList.getGroups();
+            protected Splittable convertFrom(String object) {
+                return StringQuoter.split(object);
             }
         });
```

Comment:
This is matter of preference but all these 3 methods can be condensed into  1 method if we do
```java
public void getTeams(boolean myTeam, String searchTerm, AsyncCallback<Splittable> callback)
```
and then add relevant query strings to the service call. Once we do that, we can collapse `getMyTeams`, `getTeams` and `SearchTeams` methods in `TeamsPresenterImpl`  into 1 method like
```java
public void getTeams(String Filter, String search Term)
```
This will reduce the code duplication across these 3 methods in `TeamsPresenterImpl`.

And once that is done, in `Teams.js`, you can just use `useEffect` like this:
```jsx

useEffect(() => {
        presenter.getMyTeams(teamFilter, searchTerm);
    }, [teamFilter, searchTerm]);
```
and eliminate the manual call to `presenter` methods from `handleTeamFilterChange` and `handleSearch `

Reply:
Ah, really good catch!  Will refactor.